OpenRA.Game/GameRules/RulesetCache.cs

```diff
@@ -101,7 +101,8 @@ public Ruleset Load(Map map = null)
 
 			var partial = files
 				.Select(s => MiniYaml.FromFile(s))
-				.Aggregate(nodes, MiniYaml.MergePartial);
+				.Append(nodes)
+				.Aggregate(MiniYaml.MergePartial);
```

Comment:
The remaining bogosity is hiding somewhere in here.  We need the map rules to be aggregated last in order for the removals to be processed properly, but this then causes the wrong sub-child value to be taken.

A better solution may be to move the combined merge+inherit logic up into MiniYaml, and then throw out this MergePartial junk.